aws_grant_user_access/src/process_event.py

```diff
@@ -22,10 +22,13 @@
 config = Config()
 
 
-def process_event(event: Dict[str, Any], context: Any) -> None:
+def process_event(event: Dict[str, Any], context: Any) -> str:
     logger = Config.configure_logging()
-
     validate(instance=event, schema=SCHEMA)
+    if not 1 <= event["approval_in_hours"] <= 8760:
+        retmsg = f"Invalid time period specified: {event['approval_in_hours']} hours. Valid input is 1-8760 hours"
+        logger.info(retmsg)
+        return retmsg
```

Comment:
might this logic be housed in a `is_valid_duration() -> bool` method? It will be easier to read and understand the intentions at first glance than processing the if statement.

Am I right in thinking that the intention of `retmsg` is to write a useful message in the logs? If correct, then pytest.caplog (https://docs.pytest.org/en/7.1.x/how-to/logging.html#caplog-fixture) is the pattern for testing log messages